spec/models/game_spec.rb

```diff
@@ -0,0 +1,137 @@
+describe Game do
+  subject { described_class.new(frame_class: frame_class, score_board: score_board) }
+
+  let(:frame) { instance_double(Frame, :frame, add_roll: nil, over?: false) }
+  let(:frame_class) { class_double(Frame, :frame_class, new: frame) }
+  let(:score_board) { class_double(ScoreBoard, :score_board, total_score: nil) }
+
+  describe '#score' do
+    it 'calls on score_board' do
+      expect(score_board).to receive(:total_score).once
+      subject.score
+    end
+  end
+
+  describe '#current_frame' do
+    it 'holds an instance of frame' do
+      expect(subject.current_frame).to be frame
+    end
+
+    it 'becomes a new frame after frame is over' do
+      allow(frame).to receive(:over?).and_return(true)
+      expect(frame_class).to receive(:new).twice
+      subject.add_roll(3)
+    end
+  end
+
+  describe '#add_roll' do
+    it 'adds roll to current frame' do
+      expect(frame).to receive(:add_roll).with(1)
+      subject.add_roll(1)
+    end
+
+    it 'raises error when game is over' do
+      10.times { subject.frames << frame }
+      expect { subject.add_roll(8) }.to raise_error GameOverError
+    end
+
+    context 'when frame is not over' do
+      it 'does not add current frame to frames' do
+        subject.add_roll(3)
+        expect(subject.frames).to be_empty
+      end
+    end
```

Comment:
Implementation again.

Reply:
Not reading this full test file, as I feel I've given you a few indications of how to know whether a test is implementation/state or behaviour.